processor/lookupprocessor/README.md

```diff
@@ -0,0 +1,47 @@
+# Lookup Processor
+This processor is used to lookup values in a csv file.
+
+## Supported pipelines
+- Logs
+- Metrics
+- Traces
+
+## How It Works
+1. This processor will load and periodically refresh a CSV file in memory.
+2. When telemetry is received, the processor checks if the configured `field` exists in the configured `context`.
+3. If the field exists and the CSV contains a matching value, all other values in that record are added to the `context` of the telemetry. The header of each value is used when adding these fields.
+
+## Configuration
```

Comment:
This can lead to non-obvious behavior (e.g. you can imagine 2 logs sharing 1 resource, if they have different IP values they'd both end up with the attributes for the second log, so the first log would have the incorrect attributes on its resource)

Reply:
There's also some complexity when it comes to the attributes -> body use case. If the body isn't already a map value, we have to decide whether or not to completely erase the string value or skip this operation altogether.

Reply:
I guess in my head I would like to use a body field for the lookup, and place the resulting values into attributes. In this example, I am checking for `body.ip` and I would like to have `service` and `region` go to attributes. Right now they go to body because my lookup field was under body.

![Screenshot from 2024-02-15 14-50-45](https://github.com/observIQ/bindplane-agent/assets/23043836/a8187a88-61fb-4dce-9f88-db4a56d925ab)

I cannot think of a reason someone would want to lookup an attribute and place the values into body, but the opposite would be a fairly common use-case.

Going to resource.attributes is complex like Brandon mentioned, so I suppose that case can be ignored for now (users could use the group by processor to "promote" attributes).

Reply:
In this example, would you eventually also move `ip` to attributes, or would you plan on leaving it in body?

Reply:
IP is in the log body so I would probably just leave it where it originated. Its just a thought, I am not a stickler on it and I think it is something that could be added later without breaking the current functionality.